lms/djangoapps/bulk_email/models.py

```diff
@@ -270,36 +270,16 @@ def create(
         """
         Create an instance of CourseEmail.
         """
+        # deferred import to prevent a circular import issue
+        from lms.djangoapps.bulk_email.api import determine_targets_for_course_email
+
         # automatically generate the stripped version of the text from the HTML markup:
         if text_message is None:
             text_message = html_to_text(html_message)
 
-        new_targets = []
```

Comment:
This logic was extracted to the `bulk_email` app's public Python `api.py` so it can be used outside of the `CourseEmail` model's `create` method.

This is used to determine the recipient groups (`Targets`) a message will be sent to. We needed to extract this logic so we could re-calculate the recipient groups if an email is edited to include new groups.